### PR TITLE
[DOCS] Makes the description clearer on how to use aggregations in an anomaly detection job

### LINKED ISSUE #52921: ML: Datafeed not correctly returning results
ES 7.6.1 (and 7.6.0) on Elastic Cloud.

Within the following example, the datafeed will not include field `host.name` as expected and [described within the documentation](https://www.elastic.co/guide/en/machine-learning/7.5/ml-configuring-aggregation.html).

```
PUT _ml/anomaly_detectors/farequote
{
  "analysis_config": {
    "bucket_span": "60m",
    "detectors": [{
      "function": "mean",
      "field_name": "responsetime",
      "by_field_name": "airline"
    }],
    "summary_count_field_name": "doc_count"
  },
  "data_description": {
    "time_field":"@timestamp"
  }
}

PUT /_ml/datafeeds/datafeed-farequote
{
  "job_id": "farequote",
  "indices": [
    "metricbeat-7.5.*"
  ],
  "query": {
    "bool": {
      "filter": [
        {
          "exists": {
            "field": "host.name"
          }
        },
        {
          "range": {
            "@timestamp": {
              "gte": "now-24h/h"
            }
          }
        }
      ]
    }
  },
  "aggregations": {
    "buckets": {
      "date_histogram": {
        "field": "@timestamp",
        "fixed_interval": "15m"
      },
      "aggregations": {
        "@timestamp": {
          "max": {
            "field": "@timestamp"
          }
        },
        "host.name": {
          "terms": {
            "field": "host.name"
          }
        }
      }
    }
  }
}

GET _ml/datafeeds/datafeed-farequote/_preview
```

Result:

```
[
  {
    "@timestamp" : 1582748098324,
    "doc_count" : 3997
  },
  {
    "@timestamp" : 1582748998323,
    "doc_count" : 4038
  },
  {
    "@timestamp" : 1582749898324,
    "doc_count" : 4108
  },
  ....
]
```

Instead, and in line with the documentation, I would expect the presence of `host.name` in each bucket.

The search endpoint confirms that this should work just fine:

Input:
```
GET metricbeat-7.5.*/_search
{
  "size": 0,
 "query": {
    "bool": {
      "filter": [
        {
          "exists": {
            "field": "host.name"
          }
        },
        {
          "range": {
            "@timestamp": {
              "gte": "now-24h/h"
            }
          }
        }
      ]
    }
  },
  "aggregations": {
    "buckets": {
      "date_histogram": {
        "field": "@timestamp",
        "fixed_interval": "15m"
      },
      "aggregations": {
        "host.name": {
          "terms": {
            "field": "host.name"
          }
        },
        "@timestamp": {
          "max": {
            "field": "@timestamp"
          }
        }
      }
    }
  }
}
```

Result:
```
{
  "took" : 196,
  "timed_out" : false,
  "_shards" : {
    "total" : 6,
    "successful" : 6,
    "skipped" : 0,
    "failed" : 0
  },
  "hits" : {
    "total" : {
      "value" : 10000,
      "relation" : "gte"
    },
    "max_score" : null,
    "hits" : [ ]
  },
  "aggregations" : {
    "buckets" : {
      "buckets" : [
        {
          "key_as_string" : "2020-02-26T20:00:00.000Z",
          "key" : 1582747200000,
          "doc_count" : 3997,
          "@timestamp" : {
            "value" : 1.582748098324E12,
            "value_as_string" : "2020-02-26T20:14:58.324Z"
          },
          "host.name" : {
            "doc_count_error_upper_bound" : 0,
            "sum_other_doc_count" : 0,
            "buckets" : [
              {
                "key" : "fkelbert-proxy",
                "doc_count" : 2446
              },
              {
                "key" : "fk",
                "doc_count" : 1551
              }
            ]
          }
        }
...
  }
}
```

### PR DESCRIPTION
**Pull request by szabosteve**
This PR clarifies how to use aggregations in an AD job.

Preview: http://elasticsearch_53103.docs-preview.app.elstc.co/guide/en/machine-learning/master/ml-configuring-aggregation.html

Related issue: https://github.com/elastic/elasticsearch/issues/52921

### CODE REVIEW

File: docs/reference/ml/anomaly-detection/aggregations.asciidoc [hunk 1]
```diff
@@ -11,6 +11,11 @@ distributes these calculations across your cluster. You can then feed this
 aggregated data into the {ml-features} instead of raw results, which
 reduces the volume of data that must be considered while detecting anomalies.
 
+The aggregation you create in a {dfeed} needs to have the same name as that of
```
Comment (lcawl) [inline docs/reference/ml/anomaly-detection/aggregations.asciidoc]: These seem like quite detailed requirements and I'd expect them to appear lower in the page where we're actually providing implementation examples. For example:

> The aggregation you create... the field it operates on.

I'd suggest putting this right near (or as a footnote on) the PUT _ml/datafeeds/datafeed-farequote example.

> The job analyzes only the fields that you provide in the configuration.

I'd suggest putting this right near (or as a footnote on) the PUT _ml/anomaly_detectors/farequote example.

Comment (szabosteve) [reply docs/reference/ml/anomaly-detection/aggregations.asciidoc]: Thanks! I moved the explanation to the implementation examples as suggested.

File: docs/reference/ml/anomaly-detection/aggregations.asciidoc [hunk 2]
```diff
@@ -60,6 +60,11 @@ PUT _ml/anomaly_detectors/farequote
 In this example, the `airline`, `responsetime`, and `time` fields are
 aggregations.
 
+****
```
Comment (lcawl) [inline docs/reference/ml/anomaly-detection/aggregations.asciidoc]: > Only the aggregated fields...

In my  opinion, this should be either a normal paragraph or added to the

File: docs/reference/ml/anomaly-detection/aggregations.asciidoc [hunk 3]
```diff
@@ -111,6 +116,11 @@ field is also `time`. The same is true for the aggregations with the names
 create the {dfeed}, synchronizing your aggregation and field names can simplify
 these configuration steps.
 
+****
+The aggregation you create in a {dfeed} needs to have the same name as that of
```
[the remark this answers is not in the document]
Comment (szabosteve) [reply docs/reference/ml/anomaly-detection/aggregations.asciidoc]: @dimitris-athanasiou Can we have your input here, please?
The new piece says that the aggregation name should be the same as the field name, while the IMPORTANT note says "if the aggregation name doesn't match the field name, there are limitations in the drill-down functionality within the {ml} page...". Based on this sentence, matching names are not mandatory.

**Comment by szabosteve** [comment]
@elasticmachine test this please

File: docs/reference/ml/anomaly-detection/aggregations.asciidoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (dimitris-athanasiou) [reply docs/reference/ml/anomaly-detection/aggregations.asciidoc]: It does not need to. For example, a user might set an aggregation that is then used by a pipeline aggregation. However, only the ones that have names that match the job configuration will be fed to the job.

[the remark this answers is not in the document]
Comment (lcawl) [reply docs/reference/ml/anomaly-detection/aggregations.asciidoc]: I've drafted alternative text for the "IMPORTANT" clause.  If that's still not quite correct, please just let us know, @dimitris-athanasiou. Thanks!

**Review by dimitris-athanasiou — APPROVED** [review]
LGTM

**Comment by szabosteve** [comment]
**Backport commits:**
7.x: aafc040
7.6: bf774f6

**Pull request merged**